sherpa-onnx/csrc/hypothesis.cc

```diff
@@ -17,11 +17,6 @@ void Hypotheses::Add(Hypothesis hyp) {
     hyps_dict_[key] = std::move(hyp);
   } else {
     it->second.log_prob = LogAdd<double>()(it->second.log_prob, hyp.log_prob);
-
-    if (it->second.lm_log_prob != 0 && hyp.lm_log_prob != 0) {
```

Comment:
Is there a reason to remove it?

Reply:
here two decoding paths that have the same prefix and we need to merge them into one path,
we can refer the Python implementation at here :https://github.com/k2-fsa/icefall/blob/2b3c5d799f3a585dc22071a9148424ff77aefd47/egs/librispeech/ASR/pruned_transducer_stateless2/beam_search.py#L843

python implementation perform (path1_am_prob * path1_lm_prob ) + (path2_am_prob * path2_lm_prob )
Since paths with the same prefix must have the same language model probability, we can simplify the calculation to  【path1_lm_prob * ( path1_am_prob + path2_am_prob )】

If we don't remove it , we actually perform (path1_am_prob + path2_am_prob ) * ( path1_lm_prob + path2_lm_prob) 
simplified to 【 2 * path1_lm_prob * ( path1_am_prob + path2_am_prob )】

I have tested two opensource test set and compared result :



  | 双人对话 | 中英混句
-- | -- | --
lm-shallow_fusion_0.1 | 15.18% | 22.26%
lm-shallow_fusion_0.1_bug_fix | 15.03% | 21.99%


test set reference ：
https://www.openslr.org/123/
https://magichub.com/cn/competition/code-switching-asr-challenge/

Reply:
Thanks for the detailed explanation!